mdtraj/formats/xyzfile.py

```diff
@@ -0,0 +1,350 @@
+
+        if n_frames is None:
+            frame_counter = itertools.count()
+        else:
+            frame_counter = xrange(n_frames)
+
+        if stride is None:
+            stride = 1
+
+        all_coords = []
+        for i in frame_counter:
+            try:
+                frame_coords = self._read()
+                if atom_indices is not None:
+                    frame_coords = frame_coords[atom_indices, :]
+            except _EOF:
+                break
+
+            all_coords.append(frame_coords)
+
+            for j in range(stride - 1):
+                # throw away these frames
+                try:
+                    self._read()
+                except _EOF:
+                    break
+
+        all_coords = np.array(all_coords)
+        return all_coords
+
+    def _read(self):
+        """Read a single frame. """
+
+        first = self._fh.readline()  # Number of atoms.
+        if first == '':
+            raise _EOF()
+        else:
+            self._n_atoms = int(first)
+        self._fh.readline()  # Comment line.
+        self._line_counter += 2
+
+        xyz = np.empty(shape=(self._n_atoms, 3))
+        types = np.empty(shape=self._n_atoms, dtype=str)
+
+        for i in xrange(self._n_atoms):
+            line = self._fh.readline()
+            if line == '':
+                raise _EOF()
+            split_line = line.split()
+            try:
+                types[i] = split_line[0]
+                xyz[i] = [float(x) for x in split_line[1:4]]
+            except Exception:
+                raise IOError('xyz parse error on line {0:d} of "{1:s}". '
+                              'This file does not appear to be a valid '
+                              'xyz file.'.format(
+                        self._line_counter,  self._filename))
+            self._line_counter += 1
+        # --- end body ---
+
+        self._frame_index += 1
+        return xyz
+
+    def write(self, xyz, types=None):
+        """Write one or more frames of data to a xyz file.
+
+        Parameters
+        ----------
+        xyz : np.ndarray, shape=(n_frames, n_atoms, 3)
+            The cartesian coordinates of the atoms to write.
+        types : np.ndarray, shape(3, )
+            The type of each particle.
+        """
+        if not self._mode == 'w':
+            raise ValueError('write() is only available when file is opened '
+                             'in mode="w"')
+
+        if not types:
+            # Make all particles the same type.
+            types = ['X' for _ in xrange(xyz.shape[1])]
+
+        in_units_of(xyz, 'nanometers', self.distance_unit, inplace=True)
+
+        for i in range(xyz.shape[0]):
+            self._fh.write('{0}\n'.format(xyz.shape[1]))
+            self._fh.write("Created with MDTraj {0}, {1}".format(version, str(date.today())))
+
+            for j, coord in enumerate(xyz[i]):
+                self._fh.write('{0} {1:8.3f} {2:8.3f} {3:8.3f}\n'.format(
```

Comment:
is there are particular standard documented somewhere that this format adheres to that we can link to that specifies things like the level of precision?